commands/droplet_actions.go

```diff
@@ -242,8 +257,17 @@ func RunDropletActionEnableBackups(c *CmdConfig) error {
 			return nil, err
 		}
 
-		a, err := das.EnableBackups(id)
-		return a, err
+		var defaultPolicyOnEnablingBackups = &godo.DropletBackupPolicyRequest{
```

Comment:
I had to check if default backup policy was applied to keep request body the same as before:
`{"type":"enable_backups"}` instead of having a new body - `{"backup_policy": {"hour":0, "plan":"daily", "weekday":""},"type":"enable_backups"}`

You can see changes in integration test for reference: 
[841b2f2](https://github.com/digitalocean/doctl/pull/1609/commits/841b2f2dfadf981c01cdd98cdb28af52c896cdf3) and [8f9e0e0](https://github.com/digitalocean/doctl/pull/1609/commits/8f9e0e0a9aad6e3af3fd1197252bc927116596b8)
These changes in integration test were reverted and this check was added for droplet actions enable backups.

Reply:
Let's not set our own default plan here and allow the API default to be used. It allows us to simplify things and not have to make changes if there is a decision to change the default server-side.

```diff
diff --git a/commands/droplet_actions.go b/commands/droplet_actions.go
index 90208242..fa8f9399 100644
--- a/commands/droplet_actions.go
+++ b/commands/droplet_actions.go
@@ -14,8 +14,6 @@ limitations under the License.
 package commands
 
 import (
-	"reflect"
-
 	"github.com/digitalocean/doctl"
 	"github.com/digitalocean/doctl/commands/displayers"
 	"github.com/digitalocean/doctl/do"
@@ -75,7 +73,7 @@ You can use Droplet actions to perform tasks on a Droplet, such as rebooting, re
 	cmdDropletActionEnableBackups := CmdBuilder(cmd, RunDropletActionEnableBackups,
 		"enable-backups <droplet-id>", "Enable backups on a Droplet", `Enables backups on a Droplet. This automatically creates and stores a disk image of the Droplet. By default, backups happen daily.`, Writer,
 		displayerType(&displayers.Action{}))
-	AddStringFlag(cmdDropletActionEnableBackups, doctl.ArgDropletBackupPolicyPlan, "", "daily", `Backup policy frequency plan.`)
+	AddStringFlag(cmdDropletActionEnableBackups, doctl.ArgDropletBackupPolicyPlan, "", "", `Backup policy frequency plan.`)
 	AddStringFlag(cmdDropletActionEnableBackups, doctl.ArgDropletBackupPolicyWeekday, "", "", `Backup policy weekday.`)
 	AddIntFlag(cmdDropletActionEnableBackups, doctl.ArgDropletBackupPolicyHour, "", 0, `Backup policy hour.`)
 	AddBoolFlag(cmdDropletActionEnableBackups, doctl.ArgCommandWait, "", false, "Wait for action to complete")
@@ -90,9 +88,9 @@ You can use Droplet actions to perform tasks on a Droplet, such as rebooting, re
 	cmdDropletActionChangeBackupPolicy := CmdBuilder(cmd, RunDropletActionChangeBackupPolicy,
 		"change-backup-policy <droplet-id>", "Change backup policy on a Droplet", `Changes backup policy for a Droplet with enabled backups.`, Writer,
 		displayerType(&displayers.Action{}))
-	AddStringFlag(cmdDropletActionChangeBackupPolicy, doctl.ArgDropletBackupPolicyPlan, "", "daily", `Backup policy frequency plan.`, requiredOpt())
+	AddStringFlag(cmdDropletActionChangeBackupPolicy, doctl.ArgDropletBackupPolicyPlan, "", "", `Backup policy frequency plan.`, requiredOpt())
 	AddStringFlag(cmdDropletActionChangeBackupPolicy, doctl.ArgDropletBackupPolicyWeekday, "", "", `Backup policy weekday.`)
-	AddIntFlag(cmdDropletActionChangeBackupPolicy, doctl.ArgDropletBackupPolicyHour, "", 0, `Backup policy hour.`, requiredOpt())
+	AddIntFlag(cmdDropletActionChangeBackupPolicy, doctl.ArgDropletBackupPolicyHour, "", 0, `Backup policy hour.`)
 	AddBoolFlag(cmdDropletActionChangeBackupPolicy, doctl.ArgCommandWait, "", false, "Wait for action to complete")
 	cmdDropletActionChangeBackupPolicy.Example = `The following example changes backup policy on a Droplet with the ID ` + "`" + `386734086` + "`" + `: doctl compute droplet-action change-backup-policy 386734086 --backup-policy-plan weekly --backup-policy-weekday SUN --backup-policy-hour 4`
 
@@ -257,13 +255,8 @@ func RunDropletActionEnableBackups(c *CmdConfig) error {
 			return nil, err
 		}
 
-		var defaultPolicyOnEnablingBackups = &godo.DropletBackupPolicyRequest{
-			Plan: "daily",
-			Hour: godo.PtrTo(0),
-		}
-
 		policy, err := readDropletBackupPolicy(c)
-		if err == nil && policy != nil && !reflect.DeepEqual(policy, defaultPolicyOnEnablingBackups) {
+		if err == nil && policy != nil {
 			return das.EnableBackupsWithPolicy(id, policy)
 		}
 ```

Reply:
:+1: Thanks, Andrew! I set these defaults and copied them over—I just didn’t notice them. :raised_hands: